File: consumer/ios/Nammayatri/NativeBridge/GeoCoderModule.m
//
//  GeoCoderModule.m
//  Nammayatri
//
//  Created by Khuzema Khomosi on 23/05/24.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(GeoCoderModule, NSObject)

RCT_EXTERN_METHOD(getLocName: (double)latitude
                  longitude: (double)longitude
                  resolver: (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(getGeoCoordinateFromAddress: (NSString *)address
                  resolver: (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(getAddressTranslation: (NSString *)address
                  resolver: (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject)

@end

File: consumer/ios/Nammayatri/NativeBridge/GeoCoderModule.swift
//
//  GeoCoderModule.swift
//  Nammayatri
//
//  Created by Khuzema Khomosi on 23/05/24.
//

import Foundation

@objc(GeoCoderModule)
class GeoCoderModule: NSObject {
    
    @objc
    static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    @objc
    func getLocName(_ latitude: Double, longitude: Double, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        reject("NO_LOCATION_FOUND", "Location lookup not implemented", nil)
    }
    
    @objc
    func getGeoCoordinateFromAddress(_ address: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        reject("NO_LOCATION_FOUND", "Location lookup not implemented", nil)
    }
    
    @objc
    func getAddressTranslation(_ address: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        reject("NO_LOCATION_FOUND", "Location lookup not implemented", nil)
    }
}
